src/core/symbology/qgsfillsymbollayer.cpp

```diff
@@ -4397,9 +4397,6 @@ QColor QgsCentroidFillSymbolLayer::color() const
 void QgsCentroidFillSymbolLayer::startRender( QgsSymbolRenderContext &context )
 {
   mMarker->startRender( context.renderContext(), context.fields() );
-
-  mCurrentFeatureId = -1;
```

Comment:
That's deliberate?

Reply:
Yeah, it's not used anywhere. I just noticed it while using that other class as a model.

Reply:
@nyalldawson , OK, well, merge at will :)